validator/rpc/standard_api.go

```diff
@@ -108,23 +125,37 @@ func (s *Server) ImportKeystores(
 	return &ethpbservice.ImportKeystoresResponse{Statuses: statuses}, nil
 }
 
+func groupErrorsImport(req *ethpbservice.ImportKeystoresRequest, errorMessage string) []*ethpbservice.ImportedKeystoreStatus {
+	statuses := make([]*ethpbservice.ImportedKeystoreStatus, len(req.Keystores))
+	for i := 0; i < len(req.Keystores); i++ {
+		statuses[i] = &ethpbservice.ImportedKeystoreStatus{
+			Status:  ethpbservice.ImportedKeystoreStatus_ERROR,
+			Message: errorMessage,
+		}
+	}
+	return statuses
+}
+
 // DeleteKeystores allows for deleting specified public keys from Prysm.
 func (s *Server) DeleteKeystores(
 	ctx context.Context, req *ethpbservice.DeleteKeystoresRequest,
 ) (*ethpbservice.DeleteKeystoresResponse, error) {
 	if !s.walletInitialized {
-		return nil, status.Error(codes.Internal, "Wallet not ready")
+		statuses := groupErrorsExport(req, "prysm Wallet not initialized. Please create a new wallet.")
+		return &ethpbservice.DeleteKeystoresResponse{Statuses: statuses}, nil
 	}
 	if s.validatorService == nil {
-		return nil, status.Error(codes.Internal, "Validator service not ready")
+		statuses := groupErrorsExport(req, "validator service not ready")
+		return &ethpbservice.DeleteKeystoresResponse{Statuses: statuses}, nil
 	}
 	km, err := s.validatorService.Keymanager()
 	if err != nil {
-		return nil, status.Errorf(codes.Internal, "Could not get keymanager: %v", err)
+		return nil, status.Errorf(codes.Internal, "could not get keymanager: %v", err)
```

Comment:
Capitalize RPC error strings if they are top-level errors (the first word a user sees should be capitalized)